src/cdk/v2/destinations/http/utils.js

```diff
@@ -41,12 +42,59 @@ const getCustomMappings = (message, mapping) => {
   }
 };
 
-const addPathParams = (message, apiUrl) => {
+const encodeParamsObject = (params) => {
+  if (!params || typeof params !== 'object') {
+    return {}; // Return an empty object if input is null, undefined, or not an object
+  }
+  const encodedParams = {};
+  Object.keys(params).forEach((key) => {
+    if (params[key] !== undefined) {
+      encodedParams[encodeURIComponent(key)] = encodeURIComponent(params[key]);
+    }
+  });
+  return encodedParams;
+};
+
+const getValueFromJsonPath = (message, jsonPath) => {
+  let finalPath = jsonPath;
```

Comment:
we can use `get` method right why are we bringing jsonPath

Reply:
Enhanced code